mantidproject/mantid
# Stop masked data in MD workspaces being displayed in the VSI

Closes #14613 

Consistent with the 1D and 2D plotting tools, masked data should have a signal of zero in the VSI.

**To Test:**
The VSI can plot `MDHistoWorkspaces` and `MDEventWorkspaces`. You can use the following script to create one masked and one unmasked workspace of each type. Compare the masked and unmasked workspace of each type in VSI. It should be clear that one octant of the workspace is masked, and thus not displayed.

Event workspaces can also be displayed as splatter plots in VSI. Please check that the masking works for these too, as it required separate changes. These test workspaces are rather sparsely populated, so you'll probably want to increase the "Point Size" (10 works well) and "Top Percentile" threshold (>50%) of plotted values in the "Properties" tab on the left of the window.

``` python

ws_mdhw = CreateMDWorkspace(Dimensions='3', Extents='-2,2,-2,2,-2,2', Names='h,k,l',
                       Units='rlu,rlu,rlu', SplitInto='12')
FakeMDEventData(ws_mdhw, UniformParams='1000', RandomizeSignal='1')
ws_mdhw = BinMD(ws_mdhw,AlignedDim0='h,-2,2,12',AlignedDim1='k,-2,2,12',AlignedDim2='l,-2,2,12')
ws_masked_mdhw = CloneMDWorkspace(ws_mdhw)
MaskMD(Workspace=ws_masked_mdhw,Dimensions="h,k,l",Extents="-2,0,-2,0,-2,0")

ws_mdew = CreateMDWorkspace(Dimensions='3', Extents='-2,2,-2,2,-2,2', Names='h,k,l',
                       Units='rlu,rlu,rlu', SplitInto='12')
FakeMDEventData(ws_mdew, UniformParams='1000', RandomizeSignal='1')
ws_mdew = SliceMD(ws_mdew,AlignedDim0='h,-2,2,12',AlignedDim1='k,-2,2,12',AlignedDim2='l,-2,2,12')
ws_masked_mdew = CloneMDWorkspace(ws_mdew)
MaskMD(Workspace=ws_masked_mdew,Dimensions="h,k,l",Extents="-2,0,-2,0,-2,0")

```
